contracts/Utils.sol

```diff
@@ -58,11 +88,17 @@ contract Utils {
     }
 
     function getDecimals(IERC20 token) internal view returns (uint256) {
-        if (token == ETH_TOKEN_ADDRESS) return ETH_DECIMALS; // save storage access
-        uint256 tokenDecimals = decimals[token];
+        // return token decimals if handled by token switch
+        uint256 tokenDecimals = tokenSwitch(token);
```

Comment:
lets define this in return value 
line 90 returns ....

Reply:
Done.

Reply:
I mean do not define tokenDecimals in function body
can use it in return value
which anyway saves a slot on the stack.